supplemental_notebooks/2022-09-19_muddiest_points.Rmd

```diff
@@ -0,0 +1,218 @@
+By contrast, RData files retain the name of any object you save; when you load an RData file, all object(s) in that file are automatically loaded into your current R session with whatever variable names they had at the initial export.
+This can get a little dangerous and lead to unintended code bugs, because you are loading variables into your R environment without explicitly naming those variables yourself! 
+For example, imagine you have defined an object called `sce` that contains a `SingleCellExperiment` object in your R session.
+Then, you load an RData file that contains some data you want to use.
+It is entirely possible that, _without your knowledge_, that RData file could have been storing a dataset also named `sce`. 
+This would _overwrite_ and effectively delete your `sce` variable, which will end up breaking all of your code.
+
+This difference can be partly seen in the way we read RDS vs RData files, for example:
+
+```{r, eval=FALSE}
+# Read in an RDS file and save to a variable called `my_data`
+my_data <- readr::read_rds("my_rds_file.rds")
+
+
+# Read in an RDA file, but we don't save it to a variable, because
+#  whatever variable name the given data had when it was saved to the 
+#  RData file long, long ago will be created automatically in the R session
+load("my_rda_file.rda")
+```
+
+You'll notice that when using `load()`, we don't actually create a variable - this is because variables stored in the RData file are just automatically placed into your R environment, and as mentioned, that has potential to cause frustrating bugs.
+
+
+So, if you have RData files you are now interested in converting to RDS files, you can do the following:
+
+- Use `load()` to load in your RData file
+- Look at your environment pane to see the names of the variables that were loaded in. 
+ You can also use the R command `ls()` to see all defined variables in the current session.
+- Determine which objects you want to save as RDS files, and export _each separately_ as an RDS file, with either `saveRDS()` or `readr::write_rds()`.
+
+
+
+## Labeling individual points in plots
+
+Often, we wish to label points in plots, for example labeling individual points (e.g. points representing genes of interest) in a volcano plot. 
+The example code below shows you how to include labeling for a subset of points of interest.
+
+First, let's create a toy dataset to plot with.
+Here, I've taken a random subset of 10 "male_female" contrast rows from the data frame we plotted in Intro ggplot.
+
+
+```{r}
+# Create a small dataset to use here as a plotting example. This data is actually
+# a random selection of 10 "male_female" contrast rows from the data frame we 
+# plotted in Intro ggplot.
+# The tribble() function from the tibble package (tibble::tribble()) will create a
+#  tibble, which _is_ a data frame but with a couple helpful tidyverse features
+#  More info here: https://tibble.tidyverse.org/articles/tibble.html
+gene_data <- tibble::tribble(
```

Comment:
I don't _love_ introducing `tribble` here. I tend to think reinforcing the core features is better at this point than adding new syntax, so I would use a base `data.frame` here.